tests/integration/conftest.py

```diff
@@ -82,43 +128,62 @@ def port_is_listening(hostname: str, port: int, ipv6: bool) -> bool:
     s.settimeout(0.5)
     try:
         s.connect((hostname, port))
+        s.close()
         return True
     except socket.error:
         return False
 
 
 def wait_for_kafka(port: int, *, hostname: str = "127.0.0.1", wait_time: float = 20.0) -> None:
-    start_time = time.monotonic()
     bootstrap_server = f"{hostname}:{port}"
-    while True:
-        if time.monotonic() - start_time > wait_time:
-            raise Timeout(f"Could not contact kafka cluster on host {hostname}, port {port}")
+    expiration = Expiration.from_timeout(
+        msg=f"Could not contact kafka cluster on host `{bootstrap_server}`",
+        timeout=wait_time,
+    )
+
+    list_topics_successful = False
+    while not list_topics_successful:
+        expiration.raise_if_expired()
         try:
-            client = KafkaAdminClient(bootstrap_servers=bootstrap_server)
-            _ = client.list_topics()
-            return
-        # pylint: disable=broad-except
-        except Exception as e:
+            KafkaAdminClient(bootstrap_servers=bootstrap_server).list_topics()
+        except Exception as e:  # pylint: disable=broad-except
             print(f"Error checking kafka cluster: {e}")
             time.sleep(2.0)
```

Comment:
What do you think about making 2.0 a variable? I had a feeling that with a tests amount increased it will change

Reply:
Do you mean the amount of time we wait for kafka in total? This is the time we wait in between checks. The total amount of time we wait for kafka is the `wait_time` variable.

Reply:
I mean between checks. I mean why 2 seconds? Could we wait less? If we have more tests this could make whole suit very slow, no?

Reply:
Sorry! Wrong place, I was going to leave it wait_for_port, because it is called often, and 2.0 seconds could be too much. (If it is not ready for the first time - 4 seconds... etc) So, it could be tuned

Reply:
> I mean why 2 seconds?

I suppose this is totally arbitrary. I have the impression it takes something like 6 seconds for kafka to become available, perhaps that is why this value was chosen.

> Could we wait less?

Most definitely.

> If we have more tests this could make whole suit very slow, no?

At most by `2.0` seconds.

> Because wait_for_kafka will probably used in every test

This is used only for the session fixture, which runs once before any tests. It just starts a kafka process and waits for it to be available before running the tests that depend on it.

> That is why, maybe, make this value tunable

Making it tunable for this function is not as useful as it first seems.

The problem is that this function is called by fixtures only. Configuring fixtures with tunable values is really confusing for someone that is not used to it because the call graph is implicit. Fixtures describe dependencies, these is the used to build a dependency graph, the execution of these fixtures is not shown in the code but it is rather determined at run-time. Using fixtures to tune the behavior of the build becomes rather quickly untenable. I can just decrease this to another arbitrary value, say `0.3`.